src/codeu/chat/server/Server.java

```diff
@@ -15,295 +15,424 @@
+                Uuid.SERIALIZER.write(out, info.version);
+            }
+        });
+
+        // Get Server Uptime - A client wants to get the amount of time the server has been running.
+        this.commands.put(NetworkCode.SERVER_UPTIME_REQUEST, new Command() {
+            @Override
+            public void onMessage(InputStream in, OutputStream out) throws IOException {
+                Serializers.INTEGER.write(out, NetworkCode.SERVER_UPTIME_RESPONSE);
+                Time.SERIALIZER.write(out, info.startTime);
+            }
+        });
+
+    // Status Update - A client wants to get an update on all the things they're interested in.
+    // writes the following items:
+    //   1. The updates about the users being followed - A HashMap of key-value pairs where
+    //      the key is the userid and the value is a set of conversations
+    //   2. The updates about the conversations being followed - A HashMap of key-value pairs
+    //      where the key is the conversation id and the values is a set of messages
+    this.commands.put(NetworkCode.STATUS_UPDATE_REQUEST, new Command() {
       @Override
       public void onMessage(InputStream in, OutputStream out) throws IOException {
+        final Uuid userid = Uuid.SERIALIZER.read(in);
+        final HashMap<Uuid, Collection<ConversationHeader>> interestedUsers = new HashMap<Uuid, Collection<ConversationHeader>>();
+        final HashMap<Uuid, Integer> interestedConversations = new HashMap<Uuid, Integer>();
+
+        final Interests interests = view.findInterests(userid);
+        final Collection<Uuid> uuids = interests.interests;
+        final Time lastUpdate = interests.lastStatusUpdate;
+
+        final Collection<ConversationHeader> conversations = new HashSet<ConversationHeader>(view.getConversations());
+
+        for (ConversationHeader convo : conversations) {
+          Uuid owner = convo.owner;
+          if (convo.creation.compareTo(lastUpdate) >= 0 && uuids.contains(owner)) {
+            Collection<ConversationHeader> interestedConvo = interestedUsers.get(owner);
+            interestedConvo = interestedConvo == null ? new HashSet<ConversationHeader>() : interestedConvo;
+            interestedConvo.add(convo);
+            interestedUsers.put(owner, interestedConvo);
+          }
+          for (Message message = view.findMessage(view.getConversationPayload(convo.id).firstMessage);
+            message != null;
+            message = view.findMessage(message.next)) {
+            if (message.creation.compareTo(lastUpdate) >= 0) {
+              if (uuids.contains(message.author)) {
+                Collection<ConversationHeader> interestedConvo = interestedUsers.get(owner);
+                interestedConvo = interestedConvo == null ? new HashSet<ConversationHeader>() : interestedConvo;
+                interestedConvo.add(convo);
+                interestedUsers.put(owner, interestedConvo);
```

Comment:
No, I think that this works as is, because it's a user's UUID mapped to a SET of conversation ids, not a single id.